### PR TITLE
Support NaN/Inf and -str in duration_in_seconds()

### PR DESCRIPTION
**Pull request by hagenw**
This adds support for `None`, `''`, `'None'`, `'NaN'`, `'NaT'`, `np.NaN`, `pd.NA`, `pd.NaT`, `'inf'`, `np.Inf`, `'-inf'`, `-np.Inf` to `audmath.duration_in_seconds()`.
This will also help at https://github.com/audeering/audinterface/pull/97.
It also fixes currently missing support for sign in string such as `'-1'` or `'+s'`.

image

### CODE REVIEW

File: audmath/core/api.py [hunk 1]
```diff
@@ -237,10 +251,18 @@ def to_nanos(value, unit):
 
     if isinstance(duration, str):
 
+        # none/-inf/inf duration
+        if duration == '' or duration == 'none':
```
Comment (frankenjoe) [inline audmath/core/api.py:255]: Maybe: `duration.lower() == 'none'`?

File: tests/test_api.py [hunk 2]
```diff
@@ -269,6 +269,30 @@ def test_rms(x, axis, keepdims, expected):
...
+        ('none', 1000, np.NaN),
+        (np.NaN, None, np.NaN),
+        (np.NaN, 1000, np.NaN),
+        (pd.NaT, None, np.NaN),
+        (pd.NaT, 1000, np.NaN),
+        (np.timedelta64('NaT', 's'), None, np.NaN),
+        (np.timedelta64('NaT', 's'), 1000, np.NaN),
+        (pd.to_timedelta(np.NaN, 's'), None, np.NaN),
+        (pd.to_timedelta(np.NaN, 's'), 1000, np.NaN),
+        (pd.to_timedelta(pd.NaT, 's'), None, np.NaN),
```
Comment (frankenjoe) [inline tests/test_api.py:286]: I guess we can skip the `to_timedelta()` here.

Comment (hagenw) [reply tests/test_api.py]: You are right, it's the same output:

```python
>>> pd.NaT
NaT
>>> pd.to_timedelta(pd.NaT, 's')
NaT
>>> pd.to_timedelta(np.NaN, 's')
NaT
```
I removed the tests.

File: audmath/core/api.py (hunk 1 above)
Comment (hagenw) [reply audmath/core/api.py:255]: Good question, I wasn't sure what `None` should be in string representation. If you provide `''` you don't have a value which felt similar to `None` to me. But I'm also fine with removing `''`

Comment (frankenjoe) [reply audmath/core/api.py:255]: I was actually referring to the second case since:

```
>>> str(None)
'None'
```

Comment (hagenw) [reply audmath/core/api.py]: Good point, I added `lower()` and also a test for `'None'`

Comment (frankenjoe) [reply audmath/core/api.py]: What about `NaN` / `nan`?

Comment (hagenw) [reply audmath/core/api.py]: I added support for `NaN` and `NaT` and any other lower/mixed case version of those words.

Comment (frankenjoe) [reply audmath/core/api.py]: Now that we support `None` we should probably express this in the type hint and change to `duration: typing.Optional[Union[float, int, str, np.timedelta64]],`

Comment (frankenjoe) [reply audmath/core/api.py]: And maybe also mention in the docstring that inf and nan is supported.

Comment (hagenw) [reply audmath/core/api.py]: I added `typing.Optional[]` and extended the docstring with:

image

Note, the link to `pandas.NaT` is not working, even though the link to `pandas.Timedelta` works, but I did not find a way to make `pandas.NaT` work, it seems to be some weird alias.

Comment (frankenjoe) [reply audmath/core/api.py]: Great, I think the only missing case is now `''`, which is also interpreted as `numpy.nan`.

File: audmath/core/api.py [hunk 3]
```diff
@@ -281,6 +322,14 @@ def to_nanos(value, unit):
     elif duration.__class__.__name__ == 'Timedelta':
         duration = duration.total_seconds()
 
+    # handle nan/none durations
+    elif (
+            duration is None
+            or duration.__class__.__name__ == 'NaTType'
```
Comment (frankenjoe) [inline audmath/core/api.py:329]: In case we also want to support `pd.NA` we need to extend the test to:

```suggestion
            or duration.__class__.__name__ == 'NaTType'
            or duration.__class__.__name__ == 'NAType'
```

Currently, we are getting are rather strange error:

```python
>>> audmath.duration_in_seconds(pd.NA)
TypeError: boolean value of NA is ambiguous
```

File: audmath/core/api.py (hunk 1 above)
Comment (hagenw) [reply audmath/core/api.py]: I added `''` as well to the docstring now.

File: audmath/core/api.py (hunk 3 above)
Comment (hagenw) [reply audmath/core/api.py:330]: I added support for `pd.NA` now, mention it in the docstring, and added a two tests for it.

**Comment by frankenjoe** [comment]
Great, I'd say we can merge.
